addons/delivery/models/stock_picking.py

```diff
@@ -190,6 +224,20 @@ def check_packages_are_identical(self):
                 raise UserError(_('You are shipping different packaging types in the same shipment.\nPackaging Types: %s' % package_names))
         return True
 
+    @api.multi
+    @api.returns('self', lambda value: value.id)
+    def message_post(self, **kwargs):
+        """ Overrides mail_thread message_post so that we can set delivery_mail_sent to true  of boolean field value to identify email 
+        is sent and make send confirmation email button normal from primary.
+        """
+        if self.env.context.get('mark_delivery_as_sent'):
+            for picking in self:
+                if picking.state == 'done':
+                    picking.delivery_mail_sent = True
+        return super(StockPicking, self).message_post(**kwargs)
```

Comment:
we don't have other option i think.
case is : we have to add 'Send Confirmation Email' on primary(btn-primary)  after some stage change and 
once we sent mail we have to change that button to normal from primary. 

 action_send_confirmation_email open compose wizard (there may be discard.) so to identify that mail is really send or not and change button color we have to ovverride this method.